### PR TITLE
Configuration Reader

### PR DESCRIPTION
**Pull request by Saturate**
This adds a basic configuration reader with tests. It is tested according to the merge order in #1.

It can be initialized more than once and even with other settings, however each project should only need to init it once, and pass along that final configuration.

I added the configuration to the hosts tasks.

It uses the new logging, but keeps the old console.log calls for reference for now.

### CODE REVIEW

**Comment by Saturate** [comment]
Be sure to do a squash and merge, with the relevant comments left as it's has quite a lot of commits.

**Comment by Saturate** [comment]
@Volzy Are you still alive, and want to review this? ;-)

**Comment by Volzy** [comment]
Wow. Didn't see this before now. I'll review this today.

**Comment by Saturate** [comment]
22 days ago :'(

File: lib/configuration-reader.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		this.options = Object.assign(defaultOptions, options);
+
+		return this.getConfiguration();
+	}
+
+	getConfiguration() {
+		const options = this.options;
+
+		//const defaultConfigGlob = `${options.path}/default/*${options.ext}`;
+		//const localConfigGlob = `${options.path}/local/*${options.ext}`;
```
Comment (Volzy) [inline lib/configuration-reader.js]: Is this needed? Otherwise clean it up a bit please :)

File: lib/configuration-reader.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+			}
+
+			//console.log('configFiles', configFile, config);
+
+			configurations.push(config);
+			//configArrys.push(require(configFile));
+		});
+
+
+		// TODO: Sort configs to ensure correct merge order.
```
Comment (Volzy) [inline lib/configuration-reader.js]: Tests pass for default config is overwritten by environment/local config. Is this TODO obsolete? If not I'd say that this needs to be there to ensure ConfigurationReader works as intended (doesn't seem like a big thing to sort?)

File: lib/configuration-reader.js (hunk 1 above)
Comment (Saturate) [reply lib/configuration-reader.js]: Nope, it's old. Cleaned it up.

File: lib/configuration-reader.js (hunk 2 above)
Comment (Saturate) [reply lib/configuration-reader.js]: I fixed this TODO already. Forgot to remove it thanks.

**Review by Volzy — APPROVED** [review]

**Pull request merged**